PrismSampleApp/PrismSampleApp/ViewModels/LoginPageViewModel.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Prism.Commands;
+using Prism.Mvvm;
+using Prism.Navigation;
+using Xamarin.Forms;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using System.Windows.Input;
+using System.Threading.Tasks;
+using Prism.Services;
+using PrismSampleApp.Views;
+using PrismSampleApp.Services.Interfaces;
+
+namespace PrismSampleApp.ViewModels
+{
+    public class LoginPageViewModel :ViewModelBase
+    {
+        private readonly INavigationService _navigationService;
+        private readonly IPageDialogService _pageDialogService;
+        private static ILogger logger = DependencyService.Get<ILogManager>().GetLog();
+        private string _title;
+        private string _username;
+        private string _password;
+        public string Title
+        {
+            get
+            {
+                return _title;
+
+            }
+            set
+            {
+                SetProperty(ref _title,value);
+            }
+        }
+        public LoginPageViewModel(INavigationService NavigationService,IPageDialogService pageDialogService)
+        {
+            _navigationService = NavigationService;
+            Title = "Login Page";
+            _pageDialogService = pageDialogService;
+            LoginCommand = new Command(LoginCommandHandler);
+            SendMessageCommand = new Command(SendMessageCommandHandler);
+        }
+
+        public ICommand LoginCommand { get; set; }
+        public ICommand SendMessageCommand { get; set; }
+        public string Username
+        {
+            get
+            {
+                return _username;
+            }
+            set
+            {
+                SetProperty(ref _username, value);
+            }
+        }
+        public string Password
+        {
+            get
+            {
+                return _password;
+            }
+            set
+            {
+               SetProperty(ref _password , value);
+                
+            }
+        }
+        public void SendMessageCommandHandler()
+        {
+                MessagingCenter.Send<LoginPageViewModel, DateTime>(this, "tick", DateTime.Now);
```

Comment:
Move all the hardcoded strings to AppContant if it is not binding to UI if Binding UI move it to Resx file.

Reply:
moved the hardcoded strings to Appconstant